src/badges.js

```diff
@@ -0,0 +1,953 @@
+			} else {
+				console.log(`    ${chalk.white(`Badge Templates`)}`);
+
+				const maxLengthTag = remoteTemplates.badgeTemplates.reduce((max, template) => {
+					return template.tag.length > max ? template.tag.length : max;
+				}, 0);
+				const maxLengthName = remoteTemplates.badgeTemplates.reduce((max, template) => {
+					return template.name.length > max ? template.name.length : max;
+				}, 0);
+
+				remoteTemplates.badgeTemplates.map((template) => {
+					process.stdout.write(`        ${chalk.green(template.tag.padEnd(maxLengthTag + 2))}`);
+					process.stdout.write(`${chalk.gray(template.name.padEnd(maxLengthName + 2))}`);
+					process.stdout.write(
+						`Created: ${new Date(template.createdDate).toLocaleDateString()}  Updated: ${new Date(template.updatedDate).toLocaleDateString()}\n`
+					);
+				});
+				if (remoteTemplates.locations) {
+					console.log(`    ${chalk.white(`Custom Locations`)}`);
+					console.log(`        ${chalk.green('locations')}`);
+				}
+			}
+		};
+
+		try {
+			if (options.multipleSites.length) {
+				for (let i = 0; i < options.multipleSites.length; i++) {
+					const { secretKey, siteId, name } = options.multipleSites[i];
+					await list(secretKey, siteId, name);
+					if (i < options.multipleSites.length - 1) console.log();
+				}
+			} else {
+				const { secretKey } = options.options;
+				await list(secretKey, options.context.searchspring.siteId, options.context.repository.name);
+			}
+		} catch (err) {
+			console.log(chalk.red(err));
+		}
+	}
+}
+
+export async function removeBadgeTemplate(options) {
+	const { context } = options;
+	const { searchspring, repository } = context;
+	const [command, templateName] = options.args;
+
+	if (!searchspring || !context.project || !context.project.path) {
+		console.log(chalk.red(`Error: No Snap project found in ${process.cwd()}.`));
+		return;
+	}
+
+	if (!templateName) {
+		console.log(chalk.red(`Template name is required.`));
+		return;
+	}
+
+	const payload = { name: templateName };
+
+	const remove = async (secretKey) => {
+		try {
+			// using fancy terminal output replacement
+			process.stdout.write(`${chalk.green(`        ${templateName}`)} `);
+
+			const { message } = await new ConfigApi(secretKey, options.dev).archiveBadgeTemplate(payload);
+			console.log(chalk.gray.italic(message));
+		} catch (err) {
+			process.stdout.write(chalk.red.italic(' - archived failed'));
+			console.log('        ', chalk.red(err));
+		}
+
+		await wait(100);
+	};
+
+	if (options.multipleSites.length) {
+		for (let i = 0; i < options.multipleSites.length; i++) {
+			const { secretKey, siteId, name } = options.multipleSites[i];
+
+			console.log(`${chalk.white.bold(`${name} ${chalk.cyan(`(${siteId})`)}`)}`);
+			await remove(secretKey);
+		}
+	} else {
+		const { secretKey } = options.options;
+		console.log(`${chalk.white.bold(`${repository.name}`)}`);
+		await remove(secretKey);
+	}
+}
+
+function validateLocations(locations) {
```

Comment:
I really like the effort you put into this validation, this is awesome.

<img width="818" alt="image" src="https://github.com/searchspring/snapfu/assets/5060400/4cefc2c0-29d2-4b0e-b9c0-12b2f854f0bd">